vector/src/main/java/im/vector/app/core/di/SingletonModule.kt

```diff
@@ -107,7 +110,17 @@ object VectorStaticModule {
     }
 
     @Provides
-    fun providesMatrix(context: Context): Matrix {
+    fun providesMatrixConfiguration(context: Context): MatrixConfiguration {
+        return MatrixConfiguration(
+                applicationFlavor = BuildConfig.FLAVOR_DESCRIPTION,
+                roomDisplayNameFallbackProvider = VectorRoomDisplayNameFallbackProvider(context)
+        )
+    }
+
+    @Provides
+    @Singleton
+    fun providesMatrix(context: Context, configuration: MatrixConfiguration): Matrix {
+        Matrix.initialize(context, configuration)
         return Matrix.getInstance(context)
```

Comment:
Will this work since the context is not implementing `MatrixConfiguration.Provider` anymore?

https://github.com/vector-im/element-android/blob/main/matrix-sdk-android/src/main/java/org/matrix/android/sdk/api/Matrix.kt#L108

Reply:
Ah, just read again the description, sorry

Reply:
I did it :smile: having the separate method really helps clarify the ambiguity fd2d9287e7b536ac259f950ad66ecfb11ef029f8